shared/filestore/s3store.go

```diff
@@ -115,10 +119,15 @@ func (b *S3FileBackend) s3New() (*s3.Client, error) {
 		creds = credentials.NewStatic(b.accessKey, b.secretKey, "", credentials.SignatureV4)
```

Comment:
I was envisioning something like this. We just need to modify one field. `s3.DefaultTransport` already has some customizations:

```diff
 type S3FileBackendAuthError struct {
@@ -121,18 +123,22 @@ func (b *S3FileBackend) s3New() (*s3.Client, error) {
                Region: b.region,
        }

+       tr, err := s3.DefaultTransport(b.secure)
+       if err != nil {
+               return nil, err
+       }
+       if b.skipVerify {
+               tr.TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
+       }
+       opts.Transport = tr
+
        // If this is a cloud installation, we override the default transport.
        if isCloud {
-               tr, err := s3.DefaultTransport(b.secure)
-               if err != nil {
-                       return nil, err
-               }
                scheme := "http"
                if b.secure {
                        scheme = "https"
                }
                opts.Transport = &customTransport{
-                       base:   tr,
                        host:   b.endpoint,
                        scheme: scheme,
                }
```

Also, in `customTransport` we don't use the base at all. We can remove the field.